Error in data "ybm_cluster"

Using provider version 1.0.14 (TF version 1.6.0)  - While trying to retrive data using "ybm\_cluster", I recieved:

\---------------------------------------------------------------------------------------------------  
Planning failed. Terraform encountered an error while generating this plan.

╷  
│ Error: Value Conversion Error  
│   
│   with module.yb\_cluster\["d-g-db-01"\].data.ybm\_cluster.cluster,  
│   on modules/vtg-yugabyte-multireg-cluster/main.tf line 30, in data "ybm\_cluster" "cluster":  
│   30: data "ybm\_cluster" "cluster" {  
│   
│ An unexpected error was encountered trying to convert from struct value. This is always an error in the provider. Please report the following to the provider developer:  
│   
│ couldn't find type information for attribute in supplied attr.Type types.ObjectType  
\---------------------------------------------------------------------------------------------------

DEBUG PLAN OUTPUT:

2023-10-10T11:47:54.267-0600 \[DEBUG\] provider.terraform-provider-ybm\_v1.0.14: Called provider defined DataSource Read: tf\_req\_id=8c1ada7d-7469-0a1d-b63d-6f9fe7efaccf tf\_rpc=ReadDataSource @caller=github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/fwserver/server\_readdatasource.go:66 @module=sdk.framework tf\_data\_source\_type=ybm\_cluster tf\_provider\_addr=registry.terraform.io/yugabyte/ybm timestamp=2023-10-10T11:47:54.267-0600  
2023-10-10T11:47:54.268-0600 \[ERROR\] provider.terraform-provider-ybm\_v1.0.14: Response contains error diagnostic: tf\_data\_source\_type=ybm\_cluster diagnostic\_summary="Value Conversion Error" tf\_proto\_version=6.3 tf\_req\_id=8c1ada7d-7469-0a1d-b63d-6f9fe7efaccf @caller=github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto  
 diagnostic\_detail=  
 | An unexpected error was encountered trying to convert from struct value. This is always an error in the provider. Please report the following to the provider developer:  
 |   
 | couldn't find type information for attribute in supplied attr.Type types.ObjectType  
  diagnostic\_severity=ERROR tf\_provider\_addr=registry.terraform.io/yugabyte/ybm diagnostic\_attribute="AttributeName(\\"cluster\_region\_info\\").ElementKeyInt(0).AttributeName(\\"public\_access\\")" tf\_rpc=ReadDataSource timestamp=2023-10-10T11:47:54.267-0600  
2023-10-10T11:47:54.268-0600 \[ERROR\] vertex "module.yb\_cluster\[\\"d-g-db-01\\"\].data.ybm\_cluster.cluster" error: Value Conversion Error  
2023-10-10T11:47:54.268-0600 \[ERROR\] vertex "module.yb\_cluster.data.ybm\_cluster.cluster (expand)" error: Value Conversion Error  
2023-10-10T11:47:54.273-0600 \[INFO\]  backend/local: plan operation completed  

Stack trace from the terraform-provider-ybm\_v1.0.14 plugin:

panic: runtime error: invalid memory address or nil pointer dereference  
\[signal SIGSEGV: segmentation violation code=0x2 addr=0x30 pc=0x1012d6064\]

goroutine 155 \[running\]:  
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromSlice({0x1018b9720, 0x1400071c570}, {0x0?, 0x0?}, {0x101732600?, 0x140003e57b8?, 0x10179b8e0?}, {{0x14000313630?, 0x140007b9440?, 0x140007b9440?}})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/reflect/slice.go:111 +0x64  
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue({0x1018b9720, 0x1400071c570}, {0x0?, 0x0}, {0x101732600, 0x140003e57b8}, {{0x14000313630?, 0x14000328b40?, 0x0?}})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/reflect/outof.go:68 +0x560  
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromStruct({0x1018b9720, 0x1400071c570}, {0x1018bcc70?, 0x1400075a450?}, {0x10189b0a0?, 0x140003e5500?, 0x102150108?}, {{0x101e887e0?, 0x1400008d200?, 0x140003e5500?}})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/reflect/struct.go:171 +0x354  
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue({0x1018b9720, 0x1400071c570}, {0x1018bad20?, 0x1400075a450}, {0x10189b0a0, 0x140003e5500}, {{0x101e887e0?, 0x140002ab180?, 0x140002ab3f0?}})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/reflect/outof.go:54 +0x6d0  
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromPointer({0x1018b9720, 0x1400071c570}, {0x1018bad20?, 0x1400075a450?}, {0x101723320?, 0x140003e4000?, 0x10159af58?}, {{0x101e887e0?, 0x0?, 0x0?}})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/reflect/pointer.go:115 +0x1a4  
github.com/hashicorp/terraform-plugin-framework/internal/reflect.FromValue({0x1018b9720, 0x1400071c570}, {0x1018bad20?, 0x1400075a450}, {0x101723320, 0x140003e4000}, {{0x101e887e0?, 0x0?, 0x0?}})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/reflect/outof.go:82 +0x528  
github.com/hashicorp/terraform-plugin-framework/tfsdk.(\*State).Set(0x14000255600, {0x1018b9720, 0x1400071c570}, {0x101723320?, 0x140003e4000})  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/tfsdk/state.go:155 +0xc0  
github.com/yugabyte/terraform-provider-ybm/managed.dataClusterName.Read({{{0x1016b59f0?, 0x0?}, 0x0?, 0x140004886e0?}}, {\_, \_}, {{{{0x1018bd640, 0x1400075a1e0}, {0x10179bbe0, 0x1400071da10}}, ...}, ...}, ...)  
       github.com/yugabyte/terraform-provider-ybm/managed/data\_source\_cluster\_name.go:457 +0x6ec  
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(\*Server).ReadDataSource(0x14000350240, {0x1018b9720, 0x1400071c570}, 0x140000f61c0, 0x140004555e8)  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/fwserver/server\_readdatasource.go:65 +0x30c  
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(\*Server).ReadDataSource(0x14000350240, {0x1018b9720?, 0x1400071c420?}, 0x1400009efa0)  
       github.com/hashicorp/terraform-plugin-framework@v0.10.0/internal/proto6server/server\_readdatasource.go:52 +0x184  
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(\*server).ReadDataSource(0x140002ac780, {0x1018b9720?, 0x14000707c50?}, 0x140000ef180)  
       github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/tf6server/server.go:659 +0x368  
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6.\_Provider\_ReadDataSource\_Handler({0x101878c60?, 0x140002ac780}, {0x1018b9720, 0x14000707c50}, 0x140000f6150, 0x0)  
       github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/tfplugin6/tfplugin6\_grpc.pb.go:458 +0x170  
google.golang.org/grpc.(\*Server).processUnaryRPC(0x140002a61e0, {0x1018bd918, 0x140004141a0}, 0x1400027bb00, 0x140003ec000, 0x101e4a3d0, 0x0)  
       google.golang.org/grpc@v1.56.1/server.go:1337 +0xc64  
google.golang.org/grpc.(\*Server).handleStream(0x140002a61e0, {0x1018bd918, 0x140004141a0}, 0x1400027bb00, 0x0)  
       google.golang.org/grpc@v1.56.1/server.go:1714 +0x82c  
google.golang.org/grpc.(\*Server).serveStreams.func1.1()  
       google.golang.org/grpc@v1.56.1/server.go:959 +0x84  
created by google.golang.org/grpc.(\*Server).serveStreams.func1  
       google.golang.org/grpc@v1.56.1/server.go:957 +0x16c

Error: The terraform-provider-ybm\_v1.0.14 plugin crashed!  
 

My module looks like:

```
## Since we can't use the whitelist name in the cluster resource, we need to get the whitelist id from the name
data "ybm_allow_list" "whitelist_id" {
  for_each        = toset(var.cluster_allow_list_ids)
  allow_list_name = each.value
}


resource "ybm_cluster" "multi_region_cluster" {
  cluster_name        = var.cluster_name
  cloud_type          = var.cloud_type
  cluster_type        = var.cluster_type
  cluster_region_info = var.cluster_region_info
  cluster_tier        = var.cluster_tier
  #Look this up post creation
  cluster_allow_list_ids = [
    for _, allow_list in data.ybm_allow_list.whitelist_id : allow_list.allow_list_id
  ]
  database_track    = var.database_track    #Optional
  desired_state     = var.desired_state     #Optional
  restore_backup_id = var.restore_backup_id #Optional
  fault_tolerance   = var.fault_tolerance   #Optional
  node_config       = var.node_config
  backup_schedules  = var.backup_schedules #Optional
  credentials       = var.credentials
}

data "ybm_cluster" "cluster" {
  depends_on   = [ybm_cluster.multi_region_cluster]
  cluster_name = var.cluster_name
}
```

--- a/docs/data-sources/cluster.md
+++ b/docs/data-sources/cluster.md
@@ -45,6 +45,7 @@ data "ybm_cluster" "example_cluster"{
 - `credentials` (Attributes) (see [below for nested schema](#nestedatt--credentials))
 - `database_track` (String) The track of the database. Stable or Preview.
 - `desired_state` (String) The desired state of the database, Active or Paused. This parameter can be used to pause/resume a cluster.
+- `endpoints` (Attributes List) The endpoints used to connect to the cluster. (see [below for nested schema](#nestedatt--endpoints))
 - `fault_tolerance` (String) The fault tolerance of the cluster.
 - `node_config` (Attributes) (see [below for nested schema](#nestedatt--node_config))
 - `num_faults_to_tolerate` (Number) The number of domain faults the cluster can tolerate.
@@ -81,6 +82,7 @@ Read-Only:
 Read-Only:
 
 - `num_nodes` (Number)
+- `public_access` (Boolean)
 - `region` (String)
 - `vpc_id` (String)
 - `vpc_name` (String)
@@ -148,6 +150,16 @@ Read-Only:
 - `ycql_username` (String)
 - `ysql_password` (String)
 - `ysql_username` (String)
+
+
+<a id="nestedatt--endpoints"></a>
+### Nested Schema for `endpoints`
+
+Read-Only:
+
+- `accessibility_type` (String) The accessibility type of the endpoint. PUBLIC or PRIVATE.
+- `host` (String) The host of the endpoint.
+- `region` (String) The region of the endpoint.
 
 
 <a id="nestedatt--node_config"></a>

--- a/managed/data_source_cluster_name.go
+++ b/managed/data_source_cluster_name.go
@@ -72,9 +72,12 @@ func (r dataClusterNameType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Di
 						Type:     types.StringType,
 						Computed: true,
 					},
+					"public_access": {
+						Type:     types.BoolType,
+						Computed: true,
+					},
 				}),
 			},
-
 			"backup_schedules": {
 				Computed: true,
 				Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
@@ -355,6 +358,27 @@ func (r dataClusterNameType) GetSchema(_ context.Context) (tfsdk.Schema, diag.Di
 					ElemType: types.StringType,
 				},
 				Optional: true,
+				Computed: true,
+			},
+			"endpoints": {
+				Description: "The endpoints used to connect to the cluster.",
+				Attributes: tfsdk.ListNestedAttributes(map[string]tfsdk.Attribute{
+					"accessibility_type": {
+						Description: "The accessibility type of the endpoint. PUBLIC or PRIVATE.",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"host": {
+						Description: "The host of the endpoint.",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+					"region": {
+						Description: "The region of the endpoint.",
+						Type:        types.StringType,
+						Computed:    true,
+					},
+				}),
 				Computed: true,
 			},
 			"cluster_certificate": {

